daemon/src/model/cfd.rs

```diff
@@ -1276,6 +1279,273 @@ impl Cfd {
+    /// we apply the event to the aggregate producing a new aggregate (representing the latest state
+    /// `version`). To bring a cfd into a certain state version we load all events from the
+    /// database and apply them in order (order by version).
+    pub struct CfdAggregate {
+        version: u64,
+
+        // static
+        uuid: Uuid,
+        position: Position,
+        initial_price: Price,
+        leverage: Leverage,
+        settlement_time_interval_hours: Duration,
+        quantity: Usd,
+
+        // TODO: plug in once we model this
+        // pub counterparty_network_identity: Uuid,
+
+        // dynamic (based on events)
+        dlc: Option<Dlc>,
+        cet: Option<Cet>,
+        attestation: Option<Attestation>,
+
+        revoke_commit: Option<RevokedCommit>,
+
+        // TODO: Proabbly better to bundle this in some structure?
+        lock_finality: bool,
+        commit_finality: bool,
+        refund_finality: bool,
+        cet_finality: bool,
+        collaborative_settlement_finality: bool,
+
+        cet_timelock_expired: bool,
+        refund_timelock_expired: bool,
+
+        // TODO: Can we map to cfd state for now to have contained changes in the legacy code?
+        cfd_state: CfdState,
+    }
+
+    impl CfdAggregate {
+        pub fn new(
+            uuid: Uuid,
+            position: Position,
+            initial_price: Price,
+            leverage: Leverage,
+            settlement_time_interval_hours: Duration,
+            quantity: Usd,
+        ) -> Self {
+            CfdAggregate {
+                version: 1,
+                uuid,
+                position,
+                initial_price,
+                leverage,
+                settlement_time_interval_hours,
+                quantity,
+                dlc: None,
+                cet: None,
+                attestation: None,
+                revoke_commit: None,
+                lock_finality: false,
+                commit_finality: false,
+                refund_finality: false,
+                cet_finality: false,
+                collaborative_settlement_finality: false,
+                cet_timelock_expired: false,
+                refund_timelock_expired: false,
+                cfd_state: CfdState::outgoing_order_request(),
+            }
+        }
+
+        pub fn start_contract_setup(&self) {
+            todo!(
+                "should this advance state? if it does not then there is not point in having this"
+            )
+        }
+
+        // TODO: Opted for one single function that handles all commands rather than a lot of
+        // different functions.  This makes it easier to see all commands and have the logic
+        // in one place. But it is one big function again - pretty cleaned up though.
+        //  Somewhat less typing when implementing it like this, not sure if better than having
+        // separate functions...  Separate functions might be more readable...?
```

Comment:
IMO separate functions have multiple advantages:

1. Shorter
2. Less indentation because no `match`
3. Less typing on the call site
4. Less complexity: An enum suggests that there is at least one place in the code where we need to deal with `Command`s dynamically, ie store them without knowing which concrete instance it is but that is not true. We always know which command we want to execute. Modelling this using an enum is using the wrong tool of the programming language for this job.